projects/ad9656_fmc/common/ad9656_bd.tcl

```diff
@@ -0,0 +1,115 @@
+                                               $RX_SAMPLES_PER_FRAME \
+                                               $RX_SAMPLE_WIDTH
+
+ad_ip_instance axi_dmac axi_ad9656_rx_dma
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.DMA_TYPE_SRC 2
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.DMA_TYPE_DEST 0
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.CYCLIC 0
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.SYNC_TRANSFER_START 1
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.DMA_2D_TRANSFER 0
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.DMA_DATA_WIDTH_SRC [expr 32*$RX_NUM_OF_LANES]
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.MAX_BYTES_PER_BURST 256
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.AXI_SLICE_DEST false
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.AXI_SLICE_SRC false
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.DMA_DATA_WIDTH_DEST 128
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.FIFO_SIZE 32
+
+# common cores
+
+ad_ip_instance util_adxcvr util_ad9656_xcvr
+ad_ip_parameter util_ad9656_xcvr CONFIG.RX_NUM_OF_LANES $RX_NUM_OF_LANES
+ad_ip_parameter util_ad9656_xcvr CONFIG.TX_NUM_OF_LANES 0
+ad_ip_parameter util_ad9656_xcvr CONFIG.CPLL_FBDIV 2
+ad_ip_parameter util_ad9656_xcvr CONFIG.CPLL_FBDIV_4_5 5
+ad_ip_parameter util_ad9656_xcvr CONFIG.RX_OUT_DIV 1
+ad_ip_parameter util_ad9656_xcvr CONFIG.RX_CLK25_DIV 5
+ad_ip_parameter util_ad9656_xcvr CONFIG.RX_PMA_CFG 0x001E7080
+ad_ip_parameter util_ad9656_xcvr CONFIG.RX_CDR_CFG 0x0b000023ff10400020
+
+# xcvr interfaces
+
+set rx_ref_clk     rx_ref_clk_0
+
+create_bd_port -dir I $rx_ref_clk
+ad_connect  $sys_cpu_resetn util_ad9656_xcvr/up_rstn
+ad_connect  $sys_cpu_clk util_ad9656_xcvr/up_clk
+
+# Rx
+ad_connect ad9656_rx_device_clk util_ad9656_xcvr/rx_out_clk_0
+ad_xcvrcon  util_ad9656_xcvr axi_ad9656_rx_xcvr axi_ad9656_rx_jesd {} ad9656_rx_device_clk
+ad_xcvrpll $rx_ref_clk util_ad9656_xcvr/qpll_ref_clk_0
+for {set i 0} {$i < $RX_NUM_OF_LANES} {incr i} {
+  set ch [expr $i]
+  ad_xcvrpll  $rx_ref_clk util_ad9656_xcvr/cpll_ref_clk_$ch
+  ad_xcvrpll  axi_ad9656_rx_xcvr/up_pll_rst util_ad9656_xcvr/up_cpll_rst_$ch
+}
+
+# connections (adc)
+
+ad_connect  util_ad9656_xcvr/rx_out_clk_0 rx_ad9656_tpl_core/link_clk
+ad_connect  axi_ad9656_rx_jesd/rx_sof rx_ad9656_tpl_core/link_sof
+ad_connect  axi_ad9656_rx_jesd/rx_data_tdata rx_ad9656_tpl_core/link_data
+ad_connect  axi_ad9656_rx_jesd/rx_data_tvalid rx_ad9656_tpl_core/link_valid
+ad_connect  util_ad9656_xcvr/rx_out_clk_0 util_ad9656_rx_cpack/clk
+ad_connect  ad9656_rx_device_clk_rstgen/peripheral_reset util_ad9656_rx_cpack/reset
+
+ad_connect  rx_ad9656_tpl_core/adc_valid_0 util_ad9656_rx_cpack/fifo_wr_en
+ad_connect  rx_ad9656_tpl_core/adc_enable_0 util_ad9656_rx_cpack/enable_0
+ad_connect  rx_ad9656_tpl_core/adc_enable_1 util_ad9656_rx_cpack/enable_1
+ad_connect  rx_ad9656_tpl_core/adc_data_0 util_ad9656_rx_cpack/fifo_wr_data_0
+ad_connect  rx_ad9656_tpl_core/adc_data_1 util_ad9656_rx_cpack/fifo_wr_data_1
+ad_connect  rx_ad9656_tpl_core/adc_dovf util_ad9656_rx_cpack/fifo_wr_overflow
```

Comment:
The design has 4 channels and only two are connected here

Reply:
Corrected the script. There are now 4 channels connected